lib/Sema/TypeCheckType.cpp

```diff
@@ -1032,6 +1032,50 @@ static std::string getDeclNameFromContext(DeclContext *dc,
   }
 }
 
+static Type isSE0068(TypeResolution resolution, TypeResolutionOptions options) {
+  auto dc = resolution.getDeclContext();
+  ASTContext &ctx = dc->getASTContext();
+  DeclContext *nominalDC = nullptr;
+  NominalTypeDecl *nominal = nullptr;
+  if ((nominalDC = dc->getInnermostTypeContext()) &&
+      (nominal = nominalDC->getSelfNominalTypeDecl())) {
+    assert(!isa<ProtocolDecl>(nominal) && "Cannot be a protocol");
+
+    bool insideClass = nominalDC->getSelfClassDecl() != nullptr;
+    AbstractFunctionDecl *methodDecl = dc->getInnermostMethodContext();
+    bool declaringMethod = methodDecl &&
+    methodDecl->getDeclContext() == dc->getParentForLookup();
+    bool isMutablePropertyOrSubscriptOfClass = insideClass &&
+      (options.is(TypeResolverContext::PatternBindingDecl) ||
+       options.is(TypeResolverContext::FunctionResult));
+    bool isTypeAliasInClass = insideClass &&
+      options.is(TypeResolverContext::TypeAliasDecl);
+
+//    printf("%d\n", options.getBaseContext());
+
+// Causes: Assertion failed: (hasSelfMetadataParam() && "This method can only be called if the " "SILFunction has a self-metadata parameter"), function getSelfMetadataArgument, file /Volumes/Elements/swift-self/swift/include/swift/SIL/SILFunction.h, line 955.
+
+//    if (isMutablePropertyOrSubscriptOfClass) {
+//      if (auto prop = dyn_cast_or_null<ValueDecl>
+//          (dc->getInnermostDeclarationDeclContext()))
+//        if (!prop->isSettable(dc))
+//          isMutablePropertyOrSubscriptOfClass = false;
+//    }
+
+    if (((!insideClass || !declaringMethod) &&
```

Comment:
How about this:

```
if (!insideClass ||
    options.getBaseContext() == TypeResolverContext::DynamicSelfResult)
```

Reply:
It looks like a mess but all the pieces in the `if` have a purpose according to the tests.